packages/tss/src/tss.ts

```diff
@@ -79,27 +74,23 @@ export class TKeyTSS extends ThresholdKey {
    */
   constructor(args: TSSTKeyArgs) {
     super(args);
-    const { serviceProvider, storageLayer, tssTag = "default", tssKeyType = this.keyType } = args;
+    const { serviceProvider, storageLayer, tssTag = "default", keyType } = args;
 
-    if (serviceProvider.tssKeyType !== tssKeyType) {
-      throw CoreError.default(`service provider tssKeyType mismatch: ${serviceProvider.tssKeyType} !== ${tssKeyType}`);
+    if (serviceProvider.keyType !== keyType) {
+      throw CoreError.default(`service provider keyType mismatch: ${serviceProvider.keyType} !== ${keyType}`);
     }
```

Comment:
```suggestion
```
this is already checked in super.

Reply:
in super we are checking in initialize function which wont be neccessarily called